ProjectFiles/Assets/Scripts/Network/Client.cs

```diff
@@ -62,31 +65,10 @@ public void Shutdown()
 
         public void SendLocationUpdate()
         {
-            int playerID = world.ClientID;
-            Transform transform = world.GetPlayerTransform(playerID);
-            Vector3 position = transform.position;
-            Quaternion rotation = transform.rotation;
-            Vector3 velocity = world.GetPlayerVelocity(playerID);
-            Vector3 angularVelocity = world.GetPlayerAngularVelocity(playerID);
-            using (var writer = ClientNetworkEvent.ClientLocationUpdate.GetWriter(54, Allocator.Temp))
+            var locationUpdate = new ClientLocationUpdateEvent(world);
```

Comment:
Does the event need the whole world or just the player transform?

Reply:
we just give a bunch of the other events world, I think this is fine/a non-issue

Reply:
It needs the player transform, but also the velocity + angular velocity which come from the world. If we wanna keep it world-agnostic we could pass in those values directly, I don't mind

Reply:
I think there's an argument to be made for making it world agnostic, but it's probably not worth worrying about right now. Plus having it just take world removes a lot of the duplicated extracting of the player data from the world so it does neaten the code overall.